### ISSUE TITLE
Phoneme sequence padding

### ISSUE THREAD

**Issue by WorkingJack** [yl4579/StyleTTS#35]
Hi, I'm trying to train StyleTTS with custom data, and I got a little cofusion when padding the phoneme sequence.

When pre-training AuxiliaryASR, in meldataset.py the _load_tensor function pad the phoneme sequence by blank_index.
```
   def _load_tensor(self, data):
        wave_path, text, speaker_id = data
        speaker_id = int(speaker_id)
        wave, sr = sf.read(wave_path)

        # phonemize the text
        ps = self.g2p(text.replace('-', ' '))
        if "'" in ps:
            ps.remove("'")
        text = self.text_cleaner(ps)
        blank_index = self.text_cleaner.word_index_dictionary[" "]
        text.insert(0, blank_index) # add a blank at the beginning (silence)
        text.append(blank_index) # add a blank at the end (silence)
        
        text = torch.LongTensor(text)

        return wave, text, speaker_id
```

But in StyleTTS's meldataset.py the _load_tensor function just pad the phoneme sequence by 0.
```
    def _load_tensor(self, data):
        wave_path, text, speaker_id = data
        speaker_id = int(speaker_id)
        wave, sr = sf.read(wave_path)
        if wave.shape[-1] == 2:
            wave = wave[:, 0].squeeze()
        if sr != 24000:
            wave = librosa.resample(wave, sr, 24000)
            print(wave_path, sr)
            
        wave = np.concatenate([np.zeros([5000]), wave, np.zeros([5000])], axis=0)
        
        text = self.text_cleaner(text)
        
        text.insert(0, 0)
        text.append(0)
        
        text = torch.LongTensor(text)

        return wave, text, speaker_id
```

The questions are:
1. If I pre-train ASR with blank_index, should  I keep it that way in StyleTTS?
2. If the input wavs are well-trimed(with no silence at begining and end), should I still pad the phoneme?

Thank you.

**Comment by yl4579** [comment]
1. Note that the ASR model (text aligner) is trained E2E with the TTS model, so it will eventually adapt to the padding token in the TTS model. The preprocessing code between ASR and StyleTTS repo is different, and I didn't update the ASR repo for the latest change.
2. If it is well-trimmed, there is no need to pad the phoneme (and the waveform), but I do notice some boundary effect for AdaIN, which is why I always pad with leading and ending silence to mitigate this effect. Maybe it doesn't affect your dataset.

**Comment by WorkingJack** [comment]
Thanks, but since you mentioned that you didn't update ASR repo, I noticed that ASR repo's nad pitchExtractor's model.py is slightly different from the models in StyleTTS's Utils/ASR/models.py and Utils/JDC/model.py.

Should I use the model code from StyleTTS's to train ASR and pitchExtractor?

**Comment by yl4579** [comment]
No, `model.py` is modified for inference only (like removing dropout etc.), and for training you still have to use the training repo in the original repo. You only need to change the preprocessing.
